### PR TITLE
Config support rocksdb doubly skiplist to optimize reverse-scan

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

###  What have you changed?

I have added one new memtable structure `DoublySkipList` into Rocksdb, which is similar to 
 `InlineSkipList` but optimize performance of `Iterator::Prev()`, which is used in reversed range query.  I add a configuration option to enable this feature.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

Unit tests
I have also tested this structure in `RocksDB` repo by unit tests.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

Yes.  Set `enable_doubly_skiplist` in  tikv.conf to enable this feature.

###  Does this PR affect `tidb-ansible`?

No

###  Refer to a related PR or issue link (optional)

You can see the details in this PR: [https://github.com/pingcap/rocksdb/pull/116](https://github.com/pingcap/rocksdb/pull/116)

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -334,7 +339,9 @@ macro_rules! build_cf_opt {
         cf_opts.set_soft_pending_compaction_bytes_limit($opt.soft_pending_compaction_bytes_limit.0);
         cf_opts.set_hard_pending_compaction_bytes_limit($opt.hard_pending_compaction_bytes_limit.0);
         cf_opts.set_optimize_filters_for_hits($opt.optimize_filters_for_hits);
-
+        if $opt.enable_doubly_skiplist {
+            cf_opts.set_doubly_skiplist();
```
Comment (siddontang) [inline src/config.rs]: em, why not using the same style of other RocksDB functions? 

If I remember correctly, RocksDB will pass a bool var for this function.

Comment (Little-Wallace) [reply src/config.rs]: Config which is relation to memtable all use such style, like `rocksdb_options_set_doubly_skip_list_rep(rocksdb_options_t *opt)`,  [code](https://github.com/pingcap/rocksdb/blob/tikv-3.0/db/c.cc#2567).

Comment (siddontang) [reply src/config.rs]: if I mean is `set_enable_double_skiplist(bool)`

like https://github.com/pingcap/rocksdb/blob/tikv-3.0/db/c.cc#L2451

Comment (Little-Wallace) [reply src/config.rs]: I get it. There has been default value for the `memtable_factory`, which is `SkipListFactory`, and it is defined in `options.h`.  So if we use `set_enable_double_skiplist(bool enable)`, we must know what kind of class the  `memtable_factory` should be when `enable` is set to be `false`.

**Pull request merged**